db/change_cache.go

```diff
@@ -989,7 +989,14 @@ func (c *changeCache) PushSkipped(ctx context.Context, startSeq uint64, endSeq u
 		base.InfofCtx(ctx, base.KeyCache, "cannot push negative skipped sequence range to skipped list: %d %d", startSeq, endSeq)
 		return
 	}
-	c.skippedSeqs.PushSkippedSequenceEntry(NewSkippedSequenceRangeEntry(startSeq, endSeq))
+	numSeqs := (endSeq - startSeq) + 1
+	if numSeqs > MinSequencesForRange {
+		c.skippedSeqs.PushSkippedSequenceEntry(NewSkippedSequenceRangeEntry(startSeq, endSeq))
+	} else {
+		for i := startSeq; i <= endSeq; i++ {
```

Comment:
This is going to result in a separate lock acquisition for each sequence in the range.  Perhaps we want a PushSkippedSequenceEntries(startSeq, endSeq) option on SkippedSequenceSlice that gets the lock once?  That would also let us append the set of entries to the list in a single append operation, if that turns out to be more efficient.